doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -91,68 +84,84 @@ and tables:
 Flags, options and tables are identified by a series of identifiers, each with an initial
 capital letter.
 
-.. cmd::  Set @flag
+.. cmd:: {? Export } Set {+ @ident }
    :name: Set
 
-   Sets :token:`flag` on.
+   Sets the flag identified by the :n:`@ident`\s to on.
 
-.. cmd:: Unset @flag
-   :name: Unset
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
 
-   Sets :token:`flag` off.
+   .. todo shouldn't export be included in the legacy_attr NT and listed in the table
+      in the gallina/Attributes section?  And not shown here?
 
-.. cmd:: Test @flag
+   .. todo "Set Hyps Limit." gives "Bad type of value for this option: expected int, got bool."
+      I can create an issue--when do you think someone might get to this?  The issue backlog
+      is another topic for discussion/action.
```

Comment:
I don't understand what is the issue, except that this error message should be documented. Or do you mean that the error message should be clearer (like "Hyps Limit" is an option expecting a number, not a flag)?

BTW, another thing that is not currently documented but should be is the warning "There is no option {+ @ident }."

Reply:
The issue is that no value was provided for the option.  The message should be say that.

Reply:
No value is interpreted as value `true` (if you consider that flags are "just" boolean options). I agree that this is confusing. Feel free to open a PR changing the message. In any case, it should be documented.

Reply:
This in an int-valued option, not a flag.  Would you be willing to add a check for this in #11997?  Perhaps declare_*_option could pass the type (flag, bool option, int option, table) to declare_option, which could remember it so that it's easy to check the type when the value is not provided in the `Set`.  And if so, generate a better error message?

I'd rather not touch the options code while you're working on #11997.

Reply:
I think you are misunderstanding how the code works. Flags are just considered as boolean values in the code base. And `Set flag` without a value is interpreted as setting an option to `true`. Thus the "type error". But yeah, since I've started working on the option code, I'll try to do this improvement to the error message myself.

Reply:
See #12003.